extensions/wikia/Discussions/maintenance/ForumDumper.php

```diff
@@ -93,112 +110,409 @@ public function addPage( $id, $data ) {
+	 * | parent_page_id        | int(10) unsigned | NO   | PRI | NULL                |       |
+	 * | comment_id            | int(10) unsigned | NO   | PRI | NULL                |       |
+	 * | parent_comment_id     | int(10) unsigned | NO   | MUL | 0                   |       |
+	 * | last_child_comment_id | int(10) unsigned | NO   |     | 0                   |       |
+	 * | archived              | tinyint(1)       | NO   |     | 0                   |       |
+	 * | deleted               | tinyint(1)       | NO   |     | 0                   |       |
+	 * | removed               | tinyint(1)       | NO   |     | 0                   |       |
+	 * | first_rev_id          | int(10) unsigned | NO   |     | 0                   |       |
+	 * | created_at            | datetime         | NO   |     | 0000-00-00 00:00:00 |       |
+	 * | last_rev_id           | int(10) unsigned | NO   |     | 0                   |       |
+	 * | last_touched          | datetime         | NO   | MUL | 0000-00-00 00:00:00 |       |
+	 * +-----------------------+------------------+------+-----+---------------------+-------+
+	 *
+	 * Page_wikia_props
+	 * +----------+---------+------+-----+---------+-------+
+	 * | Field    | Type    | Null | Key | Default | Extra |
+	 * +----------+---------+------+-----+---------+-------+
+	 * | page_id  | int(10) | NO   | MUL | NULL    |       |
+	 * | propname | int(10) | NO   | MUL | NULL    |       |
+	 * | props    | blob    | NO   |     | NULL    |       |
+	 * +----------+---------+------+-----+---------+-------+
+	 */
+	public function getPages( $fh = null, $pageIdsFixed = [], $minIndex = -1 ) {
+		if ( $fh == null || !empty( $this->pages ) ) {
 			return $this->pages;
 		}
 
-		$display_order = 0;
-		$dbh = wfGetDB( DB_SLAVE );
-		( new \WikiaSQL() )->SELECT( "page.*, comments_index.*, IF(pp.props is NULL,concat('i:', page.page_id, ';'), pp.props) as idx" )
-			->FROM( self::TABLE_PAGE )
-			->LEFT_JOIN( self::TABLE_COMMENTS )
-			->ON( 'page_id', 'comment_id' )
-			->LEFT_JOIN( self::TABLE_PAGE_WIKIA_PROPS )
-			->AS_( 'pp' )
-			->ON( 'page.page_id', 'pp.page_id' )
-			->AND_( 'propname', WPP_WALL_ORDER_INDEX )
-			->WHERE( 'page_namespace' )
-			->IN( self::FORUM_NAMEPSACES )
-			->ORDER_BY( 'idx' )
-			->runLoop( $dbh, function ( &$pages, $row ) use ( &$display_order ) {
-				// A few of these properties were removed and do not appear on some wikis
-				foreach ( [ 'sticky', 'locked', 'protected' ] as $prop ) {
-					if ( !property_exists( $row, $prop ) ) {
-						$row->$prop = 0;
-					}
+		$pageIdsToOrder = $this->getPageIdsDisplayOrder( $pageIdsFixed, $minIndex );
+
+		$pageIds = array_keys( $pageIdsToOrder );
+		$pageIdsChunks = array_chunk( $pageIds, 500 );
+
+		foreach ( $pageIdsChunks as $part ) {
+			$inserts = [];
+			$dbh = DumpUtils::getDBSafe( DB_SLAVE );
+			$dbh->ping();
+			( new \WikiaSQL() )->SELECT( "page.*, comments_index.*" )
```

Comment:
do we really need to fetch all fields from page and comments_index tables? if not we could save some time on sending data

Reply:
I have limited those fields, thanks!